sub-packages/bionemo-esm2/src/bionemo/esm2/model/finetune/datamodule.py

```diff
@@ -15,23 +15,114 @@
 
 
 import functools
+import os
+from typing import Sequence, Tuple
 
-import pytorch_lightning as pl
+import numpy as np
+import pandas as pd
 import torch
 import torch.utils.data
 from nemo.lightning.pytorch.plugins import MegatronDataSampler
 from nemo.utils import logging
 from pytorch_lightning.utilities.types import EVAL_DATALOADERS, TRAIN_DATALOADERS
+from torch import Tensor
+from torch.utils.data import Dataset
 
 from bionemo.core.data.multi_epoch_dataset import IdentityMultiEpochDatasetWrapper, MultiEpochDatasetResampler
 from bionemo.esm2.data import tokenizer
 from bionemo.esm2.model.finetune.finetune_regressor import InMemorySingleValueDataset
 from bionemo.esm2.model.finetune.finetune_token_classifier import InMemoryPerTokenValueDataset
 from bionemo.llm.data import collate
+from bionemo.llm.data.datamodule import MegatronDataModule
+from bionemo.llm.data.types import BertSample
 from bionemo.llm.utils.datamodule_utils import infer_num_samples
 
 
-class ESM2FineTuneDataModule(pl.LightningDataModule):
+class InMemoryCSVDataset(Dataset):
+    """An in-memory dataset that tokenize strings into BertSample instances."""
+
+    def __init__(
+        self,
+        data_path: str | os.PathLike,
+        tokenizer: tokenizer.BioNeMoESMTokenizer = tokenizer.get_tokenizer(),
+        seed: int = np.random.SeedSequence().entropy,  # type: ignore
+    ):
+        """Initializes a dataset for single-value regression fine-tuning.
+
+        This is an in-memory dataset that does not apply masking to the sequence.
+
+        Args:
+            data_path (str | os.PathLike): A path to the CSV file containing sequences.
+            labels (Optional[Sequence[float | str]]): An optional sequence of labels with 1:1 mapping to sequences.
+            tokenizer (tokenizer.BioNeMoESMTokenizer, optional): The tokenizer to use. Defaults to tokenizer.get_tokenizer().
+            seed: Random seed for reproducibility. This seed is mixed with the index of the sample to retrieve to ensure
+                that __getitem__ is deterministic, but can be random across different runs. If None, a random seed is
+                generated.
+        """
+        self.sequences, self.labels = self.load_data(data_path)
+
+        self.seed = seed
+        self._len = len(self.sequences)
+        self.tokenizer = tokenizer
+
+    def __len__(self) -> int:
+        """The size of the dataset."""
+        return self._len
+
+    def __getitem__(self, index: int) -> BertSample:
+        """Obtains the BertSample at the given index."""
+        sequence = self.sequences[index]
+        tokenized_sequence = self._tokenize(sequence)
+
+        label = tokenized_sequence if len(self.labels) == 0 else self.labels[index]
+        # Overall mask for a token being masked in some capacity - either mask token, random token, or left as-is
+        loss_mask = ~torch.isin(tokenized_sequence, Tensor(self.tokenizer.all_special_ids))
+
```

Comment:
This gives me pause. We are using this Dataset for inference, but this is implying the tokenizer is doing some kind of masking/padding.

In the `ESM2FineTuneDataModule`, there are parameters for token masking and replacement, but it doesnt look like they are used. I think it would help to either document how these parameters are used (or might be used), or to remove them.  If I was using this for the first time, I would stare at the code for a while, set them to zero, and hope for the best. But I think we should either cover this in docs or not expose those parameters.

Reply:
Yeah, I agree, I forgot to remove the masking parameters that is used for training datamodule and not here. I removed them in [f6dce08](https://github.com/NVIDIA/bionemo-framework/pull/384/commits/f6dce083fb092a12a1c001dfaeb92e3bc5895055)